sdk/identity/azure-identity/src/test/java/com/azure/identity/implementation/RetryExtension.java

```diff
@@ -6,53 +6,79 @@
  */
 public class RetryExtension implements TestExecutionExceptionHandler, BeforeTestExecutionCallback {
-
     private static final int DEFAULT_MAX_RETRIES = 3;
 
     @Override
-    public void handleTestExecutionException(ExtensionContext context, Throwable throwable) throws Throwable {
-        // Fetch or initialize retry counter
-        AtomicInteger retries = getRetries(context);
+    public void handleTestExecutionException(ExtensionContext context, Throwable throwable) {
         int maxRetries = getMaxRetries(context);
 
-        if (retries.incrementAndGet() <= maxRetries) {
-            System.out.println("Retrying test: " + context.getDisplayName() + " (Attempt " + retries.get() + ")");
-            throw new TestAbortedException("Retrying test...");
-        } else {
-            System.out.println("Exceeded retries for test: " + context.getDisplayName());
-            throw throwable; // No more retries; propagate the original exception
-        }
-    }
+        executeWithRetries(() -> {
+            try {
+                return context.getTestMethod().get().invoke(context.getRequiredTestInstance());
+            } catch (IllegalAccessException | InvocationTargetException e) {
+                throw new RuntimeException(e);
+            }
+        }, maxRetries, 1000, 2);
 
-    @Override
-    public void beforeTestExecution(ExtensionContext context) throws Exception {
-        // Reset the retry counter before a new test execution
-        context.getStore(ExtensionContext.Namespace.GLOBAL).put(getTestKey(context), new AtomicInteger(0));
     }
 
-    private AtomicInteger getRetries(ExtensionContext context) {
-        return context.getStore(ExtensionContext.Namespace.GLOBAL)
-            .getOrComputeIfAbsent(getTestKey(context), key -> new AtomicInteger(0), AtomicInteger.class);
+    @Override
+    public void beforeTestExecution(ExtensionContext context) {
+        // Initialize the retry counter before a test execution starts
+        getStore(context).put(getTestKey(context), new AtomicInteger(0));
     }
 
     private int getMaxRetries(ExtensionContext context) {
         return context.getTestMethod()
-            .flatMap(method -> method.getAnnotation(Retry.class) != null
-                ? Optional.of(method.getAnnotation(Retry.class).maxRetries())
-                : Optional.empty())
+            .flatMap(method -> Optional.ofNullable(method.getAnnotation(Retry.class)).map(Retry::maxRetries))
             .orElse(DEFAULT_MAX_RETRIES);
     }
 
+    private ExtensionContext.Store getStore(ExtensionContext context) {
+        return context.getStore(ExtensionContext.Namespace.create(getClass(), context.getUniqueId()));
+    }
+
     private String getTestKey(ExtensionContext context) {
-        return context.getUniqueId();
+        return "retries-" + context.getUniqueId();
+    }
+
+    private static <T> T executeWithRetries(Supplier<T> logic, int maxRetries, long initialDelayMillis,
```

Comment:
maybe an overload here with some sensible defaults?

Reply:
Will be addressed in https://github.com/Azure/azure-sdk-for-java/pull/43150